cli/internal/env/env.go

```diff
@@ -56,7 +65,7 @@ func getEnvPairsFromPrefixes(prefixes []string, allEnvVars map[string]string) []
 func GetHashableEnvPairs(envKeys []string, envPrefixes []string) []string {
 	allEnvVars := getEnvMap()
 	hashableEnvFromKeys := getEnvPairsFromKeys(envKeys, allEnvVars)
-	hashableEnvFromPrefixes := getEnvPairsFromPrefixes(envPrefixes, allEnvVars)
+	hashableEnvFromPrefixes := getEnvPairsFromPrefixes(envPrefixes, allEnvVars, blockedEnvVarList)
```

Comment:
What if we did this filter before adding it to the set? This feels a _bit_ weird, and I feel like explicit ordering of "generation" then "filtering" is more-clear.

Reply:
Also may make it a bit easier to compose the prefixes?

Reply:
Still passing this in so we can do it all in one loop, but open to improvements.